tests/s3_tests.py

```diff
@@ -49,3 +62,10 @@ def test_multiyear():
         assert res.shape[0] == 35040  # 2x years at 30min (1998 and 1999)
         assert isinstance(dsets, list)
         assert isinstance(ghi, np.ndarray)
+
+    files = ["s3://nrel-pds-nsrdb/current/nsrdb_1998.h5",
+             "s3://nrel-pds-nsrdb/current/nsrdb_1999.h5"]
+    with xr.open_mfdataset(files, engine="rex") as ds:
```

Comment:
It seems like fsspec with multi-file works, i don't think hsds does? This PR is great work and i think we should resolve any outstanding comments and merge, but i also have near-term use case for mfdataset with hsds. Do you know how you'd do that? I can support your time.

Reply:
Discussed this morning and fixed below. Users can now use the `rex.open_mfdataset_hsds` function to open a multi-file HSDS dataset using a glob (or an explicit list of file paths, if they wish).